index.html

```diff
@@ -1615,6 +1618,179 @@ <h3>
+              "scope": "/app"   
+            }
+          }
+        </pre>
+        <pre class="example json" 
+        title="https://help.example.com/.well-known/web-app-origin-association">
+          {
+            "https://example.com/app": {
+              "scope": "/"   
+            }
+          }
+        </pre>
+        <p> 
+          The navigation scope of this app should be all URLs that are 
+          [=manifest/within scope=] of each of 
+          `https://example.com/app`, `https://example.co.uk/app`, and 
+          `https://help.example.com`. 
+        </p>
+      </section>
+      <section>
+      <h2>Processing the scope_extensions member</h2>
+      <p>
+        To <dfn>process the `scope_extensions` member</dfn>, given
+        [=ordered map=] |json:ordered map| and [=ordered map=]
+        |manifest:ordered map|:
+      </p>
+      <ol class="algorithm">
+        <li>Let |processedScopeExtensions:list| be a new [=list=].
+        </li>
+        <li>Set |manifest|["scope_extensions"] to |processedScopeExtensions|.
+        </li>
+        <li>If |json|["scope_extensions"] doesn't [=map/exist=] or
+          |json|["scope_extensions"] is not a [=list=], return.
+        </li>
+        <li>[=list/For each=] <var>entry</var> of |json|["scope_extensions"]:
+          <ol>
+            <li>If <var>entry</var>["type"] or <var>entry</var>["value"]
+              is undefined, [=iteration/continue=].
+            </li>
+            <li>If |entry|["type"] is not `origin`, 
+              [=iteration/continue=].
+            </li>
+            <li>If <var>entry</var>["value"] is not a valid [=URL=], 
+              [=iteration/continue=].
+            </li>
+            <li>Set <var>entry</var>["value"] to its own [=origin=].
+            </li>
+            <li>Validate |entry| using the steps from [=validate scope 
+              extensions=], given [=ordered map=] |entry:ordered map|, and 
+              [=URL=] |resolved_scope:URL|.
+            </li>
+            <li>If validation failed, [=iteration/continue=].
+            </li>
+            <li>
+              Set |entry|["scope"] to |resolved_scope|.
+            </li>
+            <li>[=list/Append=] <var>entry</var> to 
+              <var>processedScopeExtensions</var>.
+            </li>
+          </ol>
+        </li>
+      </ol>
+      <aside class="note">
+        `origin` is the only type of scope extension entry that is 
+        currently specified. Other types may be added in the future to
+        support different use cases. 
+      </aside>
+      </section>
+      <section>
+      <h2>Validating scope extensions</h2>
+      <p>
+        To <dfn>validate scope extension</dfn>, given [=ordered map=] 
+        |extension:ordered map|, [=URL=] |manifest_id:URL| and [=URL=] 
+        |resolved_scope:URL|:
+      </p>
+      <ol class="algorithm">
+        <li>Set |resolved_scope| to empty URL.</li>
+        <li>Let |download_url| be the result of constructing 
```

Comment:
Probably we want to resolve the url, using manifest_url as the 'base', put that into a variable

Reply:
um.... `download_url` probably, and also probably `resolved_scope` too? Basically when we make a url, if it's from a string, we have to use a spec algorithm to do it.